test/exchange.test.ts

```diff
@@ -0,0 +1,162 @@
+/*
+Copyright 2018 - 2022 The Alephium Authors
+This file is part of the alephium project.
+
+The library is free software: you can redistribute it and/or modify
+it under the terms of the GNU Lesser General Public License as published by
+the Free Software Foundation, either version 3 of the License, or
+(at your option) any later version.
+
+The library is distributed in the hope that it will be useful,
+but WITHOUT ANY WARRANTY; without even the implied warranty of
+MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+GNU Lesser General Public License for more details.
+
+You should have received a copy of the GNU Lesser General Public License
+along with the library. If not, see <http://www.gnu.org/licenses/>.
+*/
+
+import { PrivateKeyWallet } from '@alephium/web3-wallet'
+import { getSigners, getSigner } from '@alephium/web3-test'
+import {
+  Address,
+  web3,
+  ONE_ALPH,
+  DUST_AMOUNT,
+  NodeProvider,
+  isDepositALPHTransaction,
+  getDepositAddress,
+  prettifyAttoAlphAmount
+} from '@alephium/web3'
+import { waitTxConfirmed } from '@alephium/cli'
+import { randomInt } from 'crypto'
+
+class Exchange {
+  readonly wallet: PrivateKeyWallet
+  private depositTxs: { txId: string; from: Address }[]
+  private withdrawTxs: { txId: string; to: Address }[]
+  private accountBalances: Map<Address, bigint>
+
+  constructor(wallet: PrivateKeyWallet) {
+    this.wallet = wallet
+    this.depositTxs = []
+    this.withdrawTxs = []
+    this.accountBalances = new Map<Address, bigint>()
+  }
+
+  async deposit(from: PrivateKeyWallet, amount: bigint) {
+    const result = await from.signAndSubmitTransferTx({
+      signerAddress: from.address,
+      destinations: [{ address: this.wallet.address, attoAlphAmount: amount }]
+    })
+    await waitTxConfirmed(web3.getCurrentNodeProvider(), result.txId, 1, 1000)
+    this.depositTxs.push({ txId: result.txId, from: from.address })
+    const accountBalance = this.accountBalances.get(from.address)
+    if (accountBalance === undefined) {
+      this.accountBalances.set(from.address, amount)
+    } else {
+      this.accountBalances.set(from.address, accountBalance + amount)
+    }
+  }
```

Comment:
The `Exchange` should poll new blocks and find deposit transactions with the helper functions from `exchange.ts`. Maybe we should have another class `User` which only deposit ALPH to an exchange.